web-common/src/features/dashboards/selectors.ts

```diff
@@ -122,6 +126,79 @@ export function useMetricsViewTimeRange(
   );
 }
 
+/**
+ * Wrapper function that fetches full time range.
+ * Uses useExploreValidSpec unlike the above useMetricsViewTimeRange since it is more widely used.
+ *
+ * Does an additional validation where null min and max returned throws an error instead.
+ *
+ * TODO: replace usages of useMetricsViewTimeRange with this eventually
```

Comment:
I'm not sure– do we want to use this everywhere? The Luxon settings should only be set once– in the entrypoint to the dashboard. And as long as we throw the `null` error at the top-level, then we're good.

Reply:
Hmmm true. I can remove the TODO